connectors/file/translator-parquet/src/main/java/org/teiid/translator/parquet/BaseParquetExecution.java

```diff
@@ -0,0 +1,358 @@
+       }
+    }
+
+    private MessageType getFilteredSchema(MessageType schema, List<String> expectedColumnNames) {
+        List<Type> allColumns = schema.getFields();
+        List<Type> selectedColumns = new ArrayList<>();
+        for(String columnName: expectedColumnNames) {
+            selectedColumns.add(allColumns.get(schema.getFieldIndex(columnName)));
+        }
+        MessageType filteredSchema = new MessageType(schema.getName(), selectedColumns);
+        return filteredSchema;
+    }
+
+
+    private String getExtension(String name) {
+        String extension = "";
+        int i = name.lastIndexOf('.');
+        if (i > 0) {
+            extension = name.substring(i+1);
+        }
+        return extension;
+    }
+
+    private File createTempFile(InputStream parquetFileStream) throws IOException {
+        File tempFile = File.createTempFile(getClass().getSimpleName(), ".tmp");
+        tempFile.deleteOnExit();
+        try (FileOutputStream out = new FileOutputStream(tempFile)) {
+            IOUtils.copyBytes(parquetFileStream, out, 131072);
+        }
+        return tempFile;
+    }
+
+    public Group nextRow() throws TranslatorException, DataNotAvailableException {
+            try {
+                Group row = nextRowInternal();
+                if (row == null) {
+                    return null;
+                }
+                return row;
+            } catch (IOException e) {
+                throw new TranslatorException(e);
+            }
+    }
+
+    private Group nextRowInternal() throws IOException, TranslatorException {
+        while (rowIterator != null) {
+            if (pageRowCount-- <= 0) {
+                this.rowIterator = null;
+                PageReadStore nextRowGroup = this.reader.readNextRowGroup();
+                if(nextRowGroup == null){
+                    VirtualFile nextParquetFile = getNextParquetFile();
+                    if (nextParquetFile == null) {
+                        return null;
+                    }
+                    this.rowIterator = readParquetFile(nextParquetFile);
+                    continue;
+                } else{
+                    this.pageRowCount = nextRowGroup.getRowCount();
+                    this.rowIterator = columnIO.getRecordReader(nextRowGroup, new GroupRecordConverter(filteredSchema));
+                }
+            }
+            return rowIterator.read();
+        }
+
+        return null;
+    }
+
+    protected VirtualFile getNextParquetFile() {
+        while (this.parquetFiles.length > this.fileCount.get()) {
+            VirtualFile f = this.parquetFiles[this.fileCount.getAndIncrement()];
+            if ("parquet".equals(getExtension(f.getName()))) {
+                return f;
+            }
+        }
+        return null;
+    }
+
+    private String[] getPartitionedColumns(String partitionedColumns) {
+        String[] partitionedColumnList;
+        partitionedColumnList = partitionedColumns.split(",");
```

Comment:
How do we do this?

Reply:
Capture initial limitations / exceptions on the jira so that it's easier to create the documentation.  Especially without a metadata importer we should capture:

- expectations for the partitioning property format
- type expectations for partitioning columns
- supported sources - it looks like path based partitioning will not yet work against ftp
- supported types / type mapping